docs/tutorials/setting-up-python-pycharm.md

```diff
@@ -198,3 +202,69 @@ typing `m.` and pycharm will suggest all possible methods and properties.
+- Within PyCharm choose File | Settings for Windows and Linux (or PyCharm | Preferences for OS X)
+- Expand Build, Execution, Deployment
+- Click Deployment
+- Click the + icon to add a new remote server, name it whatever you like (EV3 for example)
+- Select SFTP, then set the following parameters:
+  - SFTP host: ev3dev.local (or other name if renamed or use the IP address)
+  - Port: 22 (don't change it)
+  - Root path: / (don't bother change it as we'll set the full path later)
+  - User name: robot
+  - Password: maker (hopefully you change this at some point using the passwd command btw)
+  - Save password: Yeah check it (just easier)
+- Click the Mappings tab (if blank, hit OK the come back to this spot, sometimes it requires you to save it first)
+  - Select the path for your files on the EV3, for example:
+    - Deployment path on server: /home/robot/myfolderpath
+  - Click the button at the top that says "Use this server as default"
+  - Hit OK
+- Reopen the File | Settings for Windows and Linux (or PyCharm | Preferences for OS X) area  (note we had to save the earlier step first)
+- Visit Build, Execution, Deployment --> Deployment --> Options
+- For the Upload changed files automatically to the default server select:
+   - On explicit save action
+  
+You can test this setup without executing any code.  Just make a change, save the file, and see if the file is in sync on your EV3.  Note, in order to run that test you need an SSH terminal, which you can also do from within PyCharm (see next instruction).
+
+## Setting up an SSH Terminal within PyCharm ##
+
+This tool is really just a convenience.  Instead of using a separate program, like PuTTY, MobaXterm, or a Mac Terminal, just do it from within PyCharm. To start an SSH session just use the menu option Tools --> Start SSH Session...
+
+It should prompt you for what remote server you want to use and then open an SSH connection in a new tab to your EV3.  Assuming you have completed the step above, you will see your already configured remote all ready to go.  You can even skip that selection step by setting the default Deployment server.
+- Choose File | Settings for Windows and Linux (or PyCharm | Preferences for OS X)
+- Expand Tools
+- Select SSH Terminal
+- Change Deployment server from Select server on every run to your configured server (for example EV3 if you named it that)
+
+Done! Now both of the important steps are all within PyCharm.  You write code using your host computer, getting code completion help from your local install of ev3dev-lang--python, you load your files onto the board via the Remote Server, and then run them via an SSH Terminal that is within PyCharm.  You are ready to go!
```

Comment:
This paragraph doesn't feel right at the end of an intermediate section. What do you think about putting the conclusion at the end of the page preceded by an `hr`?